scripts/test.sh

```diff
@@ -80,30 +93,20 @@ if [[ $TEST_SUITE == "unit" ]]; then
 
         go tool cover -func profile.cov
 
-        # Disabled Coveralls.io for now
-        # To submit the test coverage result to coveralls.io,
-        # use goveralls (https://github.com/mattn/goveralls)
-        # goveralls -coverprofile=profile.cov -service=travis-ci -repotoken t47LG6BQsfLwb9WxB56hXUezvwpED6D11
-        #
-        # If running inside Travis we update coveralls. We don't want his happening on Macs
-        # if [ "$TRAVIS" == "true" ]
-        # then
-        #     n=1
-        #     until [ $n -ge 6 ]
-        #     do
-        #         echo "posting to coveralls attempt $n of 5"
-        #         goveralls -v -coverprofile=profile.cov -service travis.ci -repotoken $COVERALLS_TOKEN && break
-        #         n=$[$n+1]
-        #         sleep 30
-        #     done
-        # fi
 elif [[ $TEST_SUITE == "integration" ]]; then
-        sudo service docker restart; sleep 10
-        docker run -d -p 4242:4242 --name=opentsdb opower/opentsdb:latest
-        ip=`docker inspect -f '{{ .NetworkSettings.IPAddress }}' opentsdb`
-        while ! curl --silent -G "http://${ip}:4242" 2>&1 > /dev/null ; do
-                sleep 1
-                echo -n "."
-        done
-        PULSE_OPENTSDB_HOST=$ip go test -v --tags=integration ./...
+  docker images --format "{{.Repository}}" | grep "^mozilla/heka$" || (cd "${heka_path}/docker" && ./build_docker.sh)
+  _info "starting heka container"
+  heka_id=$(docker run -d -it -p 4352:4352 -p 3242:3242 -v "${__proj_dir}/examples/tcp-docker-test.toml":/etc/heka/config.toml mozilla/heka -config /etc/heka/config.toml)
+  _debug "container id: ${heka_id}"
+  _info "waiting for heka service"
+  ip=$(docker inspect -f '{{ .NetworkSettings.IPAddress }}' "${heka_id}")
```

Comment:
This won't work if the user is using docker-machine.  

Since the port being checked `4352` is exposed you could consider parsing the IP from the env `DOCKER_HOST` which, unless I am mistaken, will work in both cases (with and without docker-machine).